### PR TITLE
Allow customizing PS1 in myEnvFun subshell

### LINKED ISSUE #5153: Customizing myEnvFun environment
What is the proper way to customize an environment created by myEnvFun? For example, running ```load-env-foo``` changes my PS1 and unset my shell aliases. This behavior is different from using ```nix-shelll```, isn't it? Because with ```nix-shell``` everything works as expected. It saves my shell environment like PS1 and aliases.

Should I simply ```source ~/.bashrc``` inside my myEnvFun environment? However, even this approach doesn't fix all "issues" (like the missing ```__git_ps1``` function to get the current git branch I'm working on)

### PR DESCRIPTION
**Pull request by bergey**
closes https://github.com/NixOS/nixpkgs/issues/5153

### CODE REVIEW

**Comment by falsifian** [comment]
I think the substitute command described [here](http://nixos.org/nixpkgs/manual/#ssec-stdenv-functions) (or one of its variants) would make your second commit unnecessary.

File: pkgs/misc/my-env/default.nix [hunk 1]
```diff
@@ -54,10 +54,19 @@
   # The result using that command should be:
   #  env-nix loaded
   and show you a shell with a prefixed prompt.
-*/
+
+  # The prompt of the new shell can be defined by setting ps1 in the
+  # arguments to myEnvFun.  The default value is
+  # ps1 = "\\n\\$NIX_MYENV_NAME:[\\u@\\h:\\w] $ "
+  # Note that 2 backslashes are needed to pass an escape which is recognized in
```
Comment (falsifian) [inline pkgs/misc/my-env/default.nix]: Minor comment: Could you put something like "The PS1 variable will be inserted *unescaped* into the initialization script wrapped with double quotes: so don't put untrusted input here; variable substitution will work; and backslashes and double quotes must be escaped.

(Is there an easy way to escape it? That would remove the $NIX_MYNV_NAME feature, but whoever is calling this function could probably just put in ${name} and have nix do the substituting.)

**Comment by Fuuzetsu** [comment]
ping

**Comment by domenkozar** [comment]
cc @marcweber

**Comment by grahamc** [comment]
Does putting `export PS1="\\n\\$NIX_MYENV_NAME:[\\u@\\h:\\w] $ "` in a shellHook cover this use case?

**Comment by joachifm** [comment]
Is this still being worked on?

**Comment by bergey** [comment]
I'm not working on this, and I haven't heard that anyone else is.

**Pull request closed without merging**